flask_mongorest/resources.py

```diff
@@ -344,6 +357,14 @@ def value_for_field(self, obj, field):
         raise UnknownFieldError
 
     def validate_request(self, obj=None):
+        if (request.method == 'PUT' and obj) or request.method == 'POST':
```

Comment:
I'm curious what tests break and whether the main repo is utilizing this method incorrectly (or if my assumption that this method is supposed to be used only with a POST or a PUT *and* an `obj` is incorrect).